clis/teliod/src/qnap.rs

```diff
@@ -0,0 +1,458 @@
+    matches!(teliod_blocking_query!(ClientCmd::IsAlive), Ok(Ok(_)))
+}
+
+fn shutdown_teliod() -> Result<(), TeliodError> {
+    if let Ok(Ok(daemon_reply)) = teliod_blocking_query!(ClientCmd::QuitDaemon) {
+        if CommandResponse::deserialize(&daemon_reply)
+            .is_ok_and(|response| response == CommandResponse::Ok)
+        {
+            return Ok(());
+        }
+    }
+    Err(TeliodError::ClientTimeoutError)
+}
+
+fn kill_teliod_process() -> Result<(), io::Error> {
+    match fs::read_to_string(PID_FILE) {
+        Ok(pid) => {
+            // result is ignored: trivial/nowhere to log
+            let _ = Command::new("kill")
+                .arg("-9")
+                .arg(pid.trim())
+                .stdout(std::process::Stdio::null())
+                .stderr(std::process::Stdio::null())
+                .status();
+            fs::remove_file(PID_FILE)
+        }
+        Err(e) => Err(e),
+    }
+}
+
+fn start_daemon() -> Response {
+    if is_teliod_running() {
+        return text_response(StatusCode::BAD_REQUEST, "Application is already running.");
+    }
+
+    let mut teliod_log_file = match fs::OpenOptions::new()
+        .create(true)
+        .write(true)
+        .truncate(true)
+        .open(TELIOD_LOG)
+    {
+        Ok(file) => file,
+        Err(_) => {
+            return text_response(
+                StatusCode::INTERNAL_SERVER_ERROR,
+                "Failed to open teliod log file.",
+            );
+        }
+    };
+    let stdout = match teliod_log_file.try_clone() {
+        Ok(file) => Stdio::from(file),
+        Err(error) => {
+            return text_response(
+                StatusCode::INTERNAL_SERVER_ERROR,
+                format!("Failed to start the application: {error}"),
+            )
+        }
+    };
+    let stderr = match teliod_log_file.try_clone() {
+        Ok(file) => Stdio::from(file),
+        Err(error) => {
+            return text_response(
+                StatusCode::INTERNAL_SERVER_ERROR,
+                format!("Failed to start the application: {error}"),
+            )
+        }
+    };
+    match Command::new("setsid")
+        .arg(TELIOD_BIN)
+        .arg("daemon")
+        .arg(TELIOD_CFG)
+        .stdout(stdout)
+        .stderr(stderr)
+        .spawn()
+    {
+        Ok(process) => {
+            let _ = teliod_log_file.write_all(format!("Process ID: {}\n", process.id()).as_bytes());
+            text_response(StatusCode::CREATED, "Application started successfully.")
+        }
+        Err(error) => text_response(
+            StatusCode::INTERNAL_SERVER_ERROR,
+            format!("Failed to start the application: {error}"),
+        ),
+    }
+}
+
+fn stop_daemon() -> Response {
+    if shutdown_teliod().is_ok() {
+        text_response(StatusCode::OK, "Application stopped successfully.")
+    } else {
+        match kill_teliod_process() {
+            Ok(_) => text_response(StatusCode::OK, "Application stopped successfully."),
```

Comment:
I think it would be best not to use the same exact message here and in the ok case.

Reply:
`kill_teliod_process` function removed